src/requestHandler.ts

```diff
@@ -267,7 +268,9 @@ export default class RequestHandler {
       return;
     }
 
-    await this.app.vault.adapter.write(path, req.body);
+    const directory = path.relative(process.cwd(), path.dirname(filepath))
+    if (!await this.app.vault.adapter.exists(directory)) await this.app.vault.adapter.mkdir(directory)
```

Comment:
I haven't checked, but can you confirm that `this.app.vault.adapter.mkdir` will create multiple path segments if multiple directories do not exist?  I mean -- say you're creating a file at `/alpha/beta/delta.md` but neither the `alpha` nor `beta` directories exist; will it create both?

Reply:
Yup
![image](https://user-images.githubusercontent.com/19922286/206622504-116268c9-0273-4591-8ca2-1533174879e5.png)

But there is a bug of obsidian. It's will create a dir that use another separator
![image](https://user-images.githubusercontent.com/19922286/206622597-8536925f-2e67-4f01-9ef2-5108573c2cc8.png)

Reply:
I've tried `replace(/[\/\\]/g, path.sep)`. But obsidian still create the non-exist dir. And can't delete caused it's not exist. Have to restart

Reply:
I _bet_ what's happening there is that, instead of it creating intermediate path segments, it's creating a file with the name you've specified, and that name just happens to include a path segment in its name. I wonder if we need to recursively build up the directory path instead?